teacherchannel/module.py

```diff
@@ -0,0 +1,248 @@
+    ):
+        """<master channel> <slave channel>
+        Create a teacherchannel relation to sync."""
+        config = TeacherChannelDB.add_channel(ctx.guild.id, master.id, slave.id)
+        if not config:
+            await ctx.reply(_(ctx, "This channel is already assigned."))
+            return
+        await self._sync(slave, master, config)
+        await ctx.reply(_(ctx, "Teacher channel relation created."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher channel relation created. Master: <#{master.id}>, slave: <#{slave.id}>.",
+        )
+
+    @commands.check(check.acl)
+    @teacherchannel_.command(name="unset", aliases=["remove", "delete"])
+    async def teacherchannel_unset(self, ctx, channel: nextcord.TextChannel):
+        """<master channel or slave channel>
+        Remove teacherchannel relation."""
+        success = TeacherChannelDB.remove_channel(ctx.guild.id, channel.id)
+        if not success:
+            await ctx.reply(_(ctx, "This channel was never assigned."))
+            return
+        await ctx.reply(_(ctx, "Teacher channel unset."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher channel relation of channel {channel.name} unset.",
+        )
+
+    @commands.check(check.acl)
+    @teacherchannel_.group(name="teacher")
+    async def teacherchannel_teacher(self, ctx):
+        """**set** | **unset** <master channel or slave channel> <teacher member>
+        Manipulation with teachers of a teacher channel."""
+        await utils.discord.send_help(ctx)
+
+    @commands.check(check.acl)
+    @teacherchannel_teacher.command(name="set", aliases=["add"])
+    async def teacherchannel_teacher_set(
+        self, ctx, channel: nextcord.TextChannel, teacher: nextcord.Member
+    ):
+        """<master or slave channel> <teacher member>
+        Mark user as a teacher of a channel. Permission sync will not apply to this member. Grants permissions to the teacher channel (slave) only."""
+        config = TeacherChannelDB.add_teacher(ctx.guild.id, channel.id, teacher.id)
+        if not config:
+            await ctx.reply(
+                _(
+                    ctx,
+                    "Could not add teacher. Either the channel is not a teacher channel or this user has already been added.",
+                )
+            )
+            return
+        if channel.id != config.slave_id:
+            channel = ctx.guild.get_channel(config.slave_id)
+        await channel.set_permissions(teacher, read_messages=True, send_messages=True)
+        await ctx.reply(_(ctx, "Teacher added."))
+        await guild_log.info(
+            ctx.author,
+            ctx.channel,
+            f"Teacher {teacher.display_name} added to teacher channel {channel.name}.",
+        )
+
+    @commands.check(check.acl)
+    @teacherchannel_teacher.command(name="unset", aliases=["remove", "delete"])
```

Comment:
Same as with `set` and `add` - `unset` for one value, `remove` for multiple values.